File: src/main/java/com/mysawit/plantation/MysawitPlantationServiceApplication.java
package com.mysawit.plantation;

import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.context.ConfigurableApplicationContext;

@SpringBootApplication
public class MysawitPlantationServiceApplication {
    public static void main(String[] args) {
        ConfigurableApplicationContext context = SpringApplication.run(MysawitPlantationServiceApplication.class, args);
        if (context.getEnvironment().getProperty("app.test.close-context", Boolean.class, false)) {
            context.close();
        }
    }
}

File: src/main/kotlin/com/mysawit/plantation/MysawitPlantationServiceApplication.kt
package com.mysawit.plantation

import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.runApplication

@SpringBootApplication
class MysawitPlantationServiceApplication

fun main(args: Array<String>) {
    val context = runApplication<MysawitPlantationServiceApplication>(*args)
    if (context.environment.getProperty("app.test.close-context", Boolean::class.java, false)) {
        context.close()
    }
}
